### PR TITLE
import ncdu spk

### PR DESCRIPTION
**Pull request by yeled**
OK it all works now.

### CODE REVIEW

File: spk/ncdu/Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+SPK_ICON = src/ncdu.png
+
+DEPENDS = cross/$(SPK_NAME)
+
+MAINTAINER = yeled
+DESCRIPTION = disk usage analyzer with an ncurses interface
+RELOAD_UI = no
+DISPLAY_NAME = ncdu
+STARTABLE = no
+CHANGELOG = "Initial import"
```
Comment (cytec) [inline spk/ncdu/Makefile]: `Initial release` ?

File: spk/ncdu/Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+SPK_NAME = ncdu
+SPK_VERS = 1.11
+SPK_REV = 0
```
Comment (cytec) [inline spk/ncdu/Makefile]: idk for sure but i think we start at  `SPK_REV = 1` \cc @Dr-Bean

**Comment by cytec** [comment]
except those 2 comments it looks good to mee

File: spk/ncdu/Makefile (hunk 1 above)
Comment (yeled) [inline spk/ncdu/Makefile]: what would you prefer to be the first changelog?

Comment (cytec) [inline spk/ncdu/Makefile]: just replace `import` with `release` :)

Comment (yeled) [inline spk/ncdu/Makefile]: you got it.

File: cross/ncdu/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/ncursesw
+
+HOMEPAGE = https://dev.yorhel.nl/ncdu
+COMMENT  = disk usage analyzer with an ncurses interface
+LICENSE  = MIT
+
+GNU_CONFIGURE = 1
+CONFIGURE_ARGS =
```
Comment (Dr-Bean) [inline cross/ncdu/Makefile]: Remove if unused please.

File: spk/ncdu/Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+SPK_NAME = ncdu
+SPK_VERS = 1.11
+SPK_REV = 1
+SPK_ICON = src/ncdu.png
+
+DEPENDS = cross/$(SPK_NAME)
+
+MAINTAINER = yeled
+DESCRIPTION = disk usage analyzer with an ncurses interface
```
Comment (Dr-Bean) [inline spk/ncdu/Makefile]: Let's start with `Disk` instead of `disk` ;)

File: spk/ncdu/src/installer.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+preinst ()
+{
+    exit 0
+}
+
+postinst ()
+{
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+    # Put the binary in the PATH
```
Comment (Dr-Bean) [inline spk/ncdu/src/installer.sh]: Nit: A newline between L19 and L20 would make sense.

**Comment by Dr-Bean** [comment]
Few more minor comments.
As for the icon, maybe use this one: https://dev.yorhel.nl/img/ncduhelp2.png? It shows the actual name, so that makes sense.

**Comment by yeled** [comment]
@Dr-Bean and @cytec all done.

**Comment by Dr-Bean** [comment]
@yeled Care to update the PR to 1.12? Should be good to merge with that.

**Comment by yeled** [comment]
done.

**Comment by caco3** [comment]
Are there any builds available yet?
I am in need for ncdu > 1.9
I fail to be able to compile it on my Synology Diskstation (DS213+, PowerPC)

**Comment by Dr-Bean** [comment]
You can't compile packages on a DS. Check the README to set up a development environment on an x86-based PC and go from there.

As for this PR: it's not merged, so we don't have packages up on the repository yet. One of these days, a maintainer will get around to merging and publishing packages.

**Comment by caco3** [comment]
@Dr-Bean I am aware that I can't use spksrc on the DS itself, but I have gcc and all required dev-packages installed there and usually build directly there. How ever ncdu fails and I couldn't jet solve this :(

Setting up a build environment on my PC AFAIK wouldn't help since ncdu is not yet merged in, right?

**Comment by Dr-Bean** [comment]
If you have a build env setup on a PC, including spksrc, you can just merge the PR locally?

**Comment by caco3** [comment]
@Dr-Bean Yes, I could, but this seems to be a bit too much work for just one application..

Anyway, somebody (@wodny) came up with a great idea and implemented a file system scanner which generates a file which then can be parsed by ncdu (on another PC): https://github.com/wodny/ncdu-export
So this works great for me.

**Comment by Dr-Bean** [comment]
Merged with 5b1539f67b1f21833bde276eedce18e8525b1e07, additional change with https://github.com/SynoCommunity/spksrc/commit/a7444d202231c3aa3371ca6217080e8983129a11.
Thanks for the addition @yeled.

**Comment by Dr-Bean** [comment]
I have published ncdu 1.12-1 on the repository.

**Comment by caco3** [comment]
Thank you very much!
It works like a charm.

**Pull request merged**